plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-order-downloads-controller.php

```diff
@@ -0,0 +1,282 @@
+<?php
+/**
+ * REST API Order Downloads controller
+ *
+ * Handles requests to the /orders/<order_id>/downloads endpoint.
+ *
+ * @package Automattic/WooCommerce/RestApi
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Order Downloads controller class.
+ *
+ * @package Automattic/WooCommerce/RestApi
+ * @extends WC_REST_Controller
+ */
+class WC_REST_Order_Downloads_Controller extends WC_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'orders/(?P<order_id>[\d]+)/downloads';
+
+	/**
+	 * Post type.
+	 *
+	 * @var string
+	 */
+	protected $post_type = 'shop_order';
+
+	/**
+	 * Register the routes for order downloads.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'order_id' => array(
+						'description' => __( 'The order ID.', 'woocommerce' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Check whether a given request has permission to read order downloads.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		$object = $this->get_object( (int) $request['id'] );
```

Comment:
The `WC_REST_Controller` parent class doesn't have a `get_object` method, so I'm getting a fatal error here. This seems kind of redundant anyway, since the existence of the specific order is checked again in the `get_items` method. Could probably do the same as in the Order Notes controller and just check the post permissions here.